cmd/clusterctl/pkg/client/config.go

```diff
@@ -47,19 +48,150 @@ func (c *clusterctlClient) GetProviderComponents(provider, targetNameSpace, watc
 	return components, nil
 }
 
+// GetClusterTemplateOptions carries the options supported by GetClusterTemplate.
+type GetClusterTemplateOptions struct {
```

Comment:
```suggestion
type GetClusterTemplateInput struct {
```

Consider this for more clarrity to users

Reply:
I have no strong opinion, but all the other types in input to high-level types use options: InitOptions, DeleteOptions, MoveOptions, PlanUpgradeOptions, so I would prefer to keep this consistent (leave as it is or change all, but in another PR)

Reply:
`Options` tells me I'm configuring something, but here and in the other places you mentioned, the structs are really just used as input to avoid a bunch of arguments. We can refactor later, but to keep it consistent with other parts of Cluster API and providers, Input/Output is preferred